azurelinuxagent/common/protocol/wire.py

```diff
@@ -220,6 +220,12 @@ def _build_health_report(incarnation, container_id, role_instance_id, # pylint:
                          status, substatus, description):
     # Escape '&', '<' and '>'
     description = saxutils.escape(ustr(description))
+
+    # The max description that can be sent to WireServer is 4096 bytes.
+    # Exceeding this max can result in a failure to report health.
+    max_description_length = 4096
+    description = description[-max_description_length:]
```

Comment:
@trstringer Please log a warning if the description field is being trimmed. Also please leave a small but generous character buffer, as the platform usually mutates the values along the way (escaping and transforming chars). Thus the limit should be much lower than 4096.

Reply:
Once walinuxagent sends out the health status, why would the platform mutate any values? We already escape characters on the walinuxagent side, not sure I understand why that would happen again on the server side?